spec/injectable_spec.rb

```diff
@@ -215,6 +215,24 @@ def call
     end
   end
 
+  context 'with dependencies that have a call: option with a constant' do
+    subject do
+      Class.new do
+        include Injectable
+        dependency(:some_constant, call: :name) { Object }
+
+        def call
+          some_constant.call
+        end
+      end
+    end
+
+    it 'does not try to patch the dependency twice' do
+      subject.call
+      expect(subject.call).to eq 'Object'
+    end
```

Comment:
I'd add a test to verify that `Object` doesn't get monkey patched outside of the context.

Something like:
```rb
expect(Object).not_to respond_to :call
```